esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1223,25 +1246,32 @@ void CEspHttpServer::logoutSession(EspAuthRequest& authReq, unsigned sessionID,
         sendMessage(nullptr, "text/html; charset=UTF-8");
 }
 
-void CEspHttpServer::handleAuthFailed(bool sessionAuth, EspAuthRequest& authReq)
+EspAuthState CEspHttpServer::handleAuthFailed(bool sessionAuth, EspAuthRequest& authReq)
 {
     ISecUser *user = authReq.ctx->queryUser();
-    if (user && (user->getAuthenticateStatus() == AS_PASSWORD_EXPIRED || user->getAuthenticateStatus() == AS_PASSWORD_VALID_BUT_EXPIRED))
+    if (user && user->getAuthenticateStatus() == AS_PASSWORD_VALID_BUT_EXPIRED)
     {
-        ESPLOG(LogMin, "ESP password expired for %s", authReq.ctx->queryUserId());
-        handlePasswordExpired(sessionAuth);
-        return;
+        ESPLOG(LogMin, "ESP password expired for %s. Asking update ...", authReq.ctx->queryUserId());
+        if (sessionAuth) //For session auth, store the userid to cookie for the updatepasswordinput form.
+            addCookie(SESSION_ID_TEMP_COOKIE, authReq.ctx->queryUserId(), 0);
+        m_response->redirect(*m_request.get(), "/esp/updatepasswordinput");
```

Comment:
should this quoted string be placed in espcontext.hpp?  Perhaps better to have them all in a common place?